tools/parquetDDL/main.cpp

```diff
@@ -0,0 +1,285 @@
+    {
+      // set 8000 as the maximum length and VARCHAR as column type
+      colType = "VARCHAR(8000) character set 'binary'";
+      break;
+    }
+    case arrow::Type::type::FIXED_SIZE_BINARY:
+    {
+      std::shared_ptr<arrow::FixedSizeBinaryType> fType = std::static_pointer_cast<arrow::FixedSizeBinaryType>(dataType);
+      int byteWidth = fType->byte_width();
+      colType = "CHAR(" + std::to_string(byteWidth) + ")";
+      break;
+    }
+    case arrow::Type::type::DATE32:
+    {
+      colType = "DATE";
+      break;
+    }
+    case arrow::Type::type::DATE64:
+    {
+      colType = "DATE";
+      break;
+    }
+    case arrow::Type::type::TIMESTAMP:
+    {
+      std::shared_ptr<arrow::TimestampType> fType = std::static_pointer_cast<arrow::TimestampType>(dataType);
+
+      if (fType->unit() == arrow::TimeUnit::MILLI)
+        colType = "TIMESTAMP(3)";
+      else if (fType->unit() == arrow::TimeUnit::MICRO)
+        colType = "TIMESTAMP(6)";
+      else
+        return UNSUPPORTED_DATA_TYPE;
+
+      break;
+    }
+    case arrow::Type::type::TIME32:
+    {
+      colType = "TIME(3)";
+      break;
+    }
+    case arrow::Type::type::TIME64:
+    {
+      std::shared_ptr<arrow::Time64Type> fType = std::static_pointer_cast<arrow::Time64Type>(dataType);
+
+      if (fType->unit() == arrow::TimeUnit::MICRO)
+        colType = "TIME(6)";
+      else
+        return UNSUPPORTED_DATA_TYPE;
+        
+      break;
+    }
+    case arrow::Type::type::DECIMAL128:
+    {
+      // get precision and scale
+      std::shared_ptr<arrow::DecimalType> fType = std::static_pointer_cast<arrow::DecimalType>(dataType);
+      int32_t fPrecision = fType->precision();
+      int32_t fScale = fType->scale();
+      colType = "DECIMAL(" + std::to_string(fPrecision) + "," + std::to_string(fScale) + ")";
+      break;
+    }
+    default:
+    {
+      return UNSUPPORTED_DATA_TYPE;
+    }
+  }
+  return NO_ERROR;
+}
+
+/**
+ * main function to generate DDL file
+*/
+int generateDDL(std::string filePath, std::string targetPath, std::string tableName)
+{
+  std::shared_ptr<arrow::Schema> parquetSchema;
+  getSchema(filePath, &parquetSchema);
+  std::vector<std::string> parquetCols;
+  std::vector<std::string> parquetTypes;
+  int rc = NO_ERROR;
+  int fieldsNum = parquetSchema->num_fields();
+
+  if (fieldsNum == 0)
+  {
+    return EMPTY_FIELD;
+  }
+
+  for (int i = 0; i < fieldsNum; i++)
+  {
+    const std::shared_ptr<arrow::Field> tField = parquetSchema->field(i);
+    const std::string tName = tField->name();
+    std::string colType;
+    auto tType = tField->type();
+    parquetCols.push_back(tName);
+    rc = convert2mcs(tType, tType->id(), colType);
+
+    if (rc != NO_ERROR)
+    {
+      std::cout << "Not allowed data type: " << tName << std::endl;
+      return rc;
+    }
+
+    parquetTypes.push_back(colType);
+  }
+
+  std::string str1 = "CREATE TABLE " + tableName + "(\n";
+  std::string str2 = ") ENGINE=Columnstore;";
```

Comment:
There is no default encoding/collation set despite the fact that we support only utf8.